src/three/math/OBB.js

```diff
@@ -17,25 +24,42 @@ export class OBB {
 		inverseTransform.copy( transform ).invert();
 
 		const { min, max } = box;
-		let index = 0;
-		for ( let x = - 1; x <= 1; x += 2 ) {
+		points[ 0 ].set( min.x, min.y, min.z ).applyMatrix4( transform ); // Front-bottom-left
+		points[ 1 ].set( max.x, min.y, min.z ).applyMatrix4( transform ); // Front-bottom-right
+		points[ 2 ].set( max.x, max.y, min.z ).applyMatrix4( transform ); // Front-top-right
+		points[ 3 ].set( min.x, max.y, min.z ).applyMatrix4( transform ); // Front-top-left
 
-			for ( let y = - 1; y <= 1; y += 2 ) {
+		points[ 4 ].set( min.x, min.y, max.z ).applyMatrix4( transform ); // Back-bottom-left
+		points[ 5 ].set( max.x, min.y, max.z ).applyMatrix4( transform ); // Back-bottom-right
+		points[ 6 ].set( max.x, max.y, max.z ).applyMatrix4( transform ); // Back-top-right
+		points[ 7 ].set( min.x, max.y, max.z ).applyMatrix4( transform ); // Back-top-left
+		this.planeNeedsUpdate = true;
 
-				for ( let z = - 1; z <= 1; z += 2 ) {
+	}
 
-					points[ index ].set(
-						x < 0 ? min.x : max.x,
-						y < 0 ? min.y : max.y,
-						z < 0 ? min.z : max.z,
-					).applyMatrix4( transform );
-					index ++;
+	updatePlanes() {
 
-				}
+		if ( ! this.planeNeedsUpdate ) return;
 
-			}
+		const normals = [
+			_normal1.subVectors( this.points[ 1 ], this.points[ 0 ] ).cross( _vec3.subVectors( this.points[ 2 ], this.points[ 0 ] ) ).normalize(),
+			_normal2.subVectors( this.points[ 4 ], this.points[ 0 ] ).cross( _vec3.subVectors( this.points[ 1 ], this.points[ 0 ] ) ).normalize(),
+			_normal3.subVectors( this.points[ 2 ], this.points[ 0 ] ).cross( _vec3.subVectors( this.points[ 4 ], this.points[ 0 ] ) ).normalize()
+		];
 
-		}
+		// Front and back
+		this.planes[ 0 ].setFromNormalAndCoplanarPoint( normals[ 0 ], this.points[ 0 ] );
+		this.planes[ 1 ].setFromNormalAndCoplanarPoint( normals[ 0 ].negate(), this.points[ 6 ] );
+
+		// Left and right
+		this.planes[ 2 ].setFromNormalAndCoplanarPoint( normals[ 1 ], this.points[ 0 ] );
+		this.planes[ 3 ].setFromNormalAndCoplanarPoint( normals[ 1 ].negate(), this.points[ 2 ] );
+
+		// Top and bottom
+		this.planes[ 4 ].setFromNormalAndCoplanarPoint( normals[ 2 ], this.points[ 0 ] );
+		this.planes[ 5 ].setFromNormalAndCoplanarPoint( normals[ 2 ].negate(), this.points[ 1 ] );
+
+		this.planeNeedsUpdate = false;
```

Comment:
I think these functions have become a bit difficult to follow. Instead of pulling points out of the points array we can generate the planes in a more explicit method that's hopefully easier to validate:

```js
_worldMin.copy( this.box.min ).applyMatrix4( this.transform );
_worldMax.copy( this.box.max ).applyMatrix4( this.transform );

_norm.set( 0, 0, 1 ).transformDirection( this.transform );
plane[ 0 ]. setFromNormalAndCoplanarPoint( _norm, _worldMin );
plane[ 1 ]. setFromNormalAndCoplanarPoint( _norm, _worldMax ).negate();

// ...
```

Note that just like the frustum planes the "positive" side of the plane will need to face "inward".

Reply:
oh I missed this one, I'll use setFromNormalAndCoplanarPoint instead